grammar/examples.py

```diff
@@ -0,0 +1,77 @@
+all_s = [
+    r"hello",
+    r"hello\ world",
+    r"hello\,world",
+    r'"hello, world"',
+    r"'hello, world'",
+    r"123",
+    r"123.1",
+    r"123.",
+    r".0",
+    r"1e-6",
+    r"'123_abc'",
+    r"'123 abc'",
+    r"'.1_abc'",
+    r"true",
+    r"null",
+    r"nulltrue",
+    r"[1,2,3]",
+    r"[1, 2, 3]",
+    r"[1, '2 ', 3]",
+    r"[1, '2 ', 'hello world']",
+    r"[[3, 4]]",
+    r"[[1, 2], 3, 4]",
+    r"[[1, 2], [3, 4]]",
+    r'{"a": 0, "b": 1}',
+    r'{null: [0, 3.14, false], true: {"a": [0, 1, 2], "b": {}}}',
+    "[10,str,3.14,true,false,nan,inf,[1,2,3], 'quoted', \"quoted\", 'a,b,c']",
+    "${foo}",
+    "${${foo}}",
+    "${${foo}.bar}",
+    "${${foo}.bar.${baz}}",
+    "${foo.bar}",
+    "${foo:bar}",
+    "${${foo}:bar,3.14,${baz:null}}",
+    "${foo:[1,2,3]}",
+    "${foo:{1:${bar}, 3: 4}}",
+    "${foo:'+'}",
+    "hello_${foo}",
+    "${foo}_hello",
+    "hello_${foo}_world",
+    "${foo}${bar}",
+    "'x='${x}',y='${y}",
+    "${foo:'x='${x},'y='${y}}",
+    "${foo:}",
+    "1_000",
+    "_1_000",
+    "0",
+    "${foo:1_000_000,3.13}",
+    "1_000",
+    "abc'def'",
+    "${a:b}",
+    ".a:b.c",
+    ":.a#b:c",
+    "a.b#c.",
+    ":a#b#c.",
+    "a:b.c",
+    "http://test:8080",
+    "a.b",
+    "http//${server.host}:${server.port}",
+    "a:b",
+    "ab_%",
+    "null${a}",
+    "null?${a}",
+    "${foo:10,str,3.14,true,false,nan,inf,[1,2,3], ${foo:bar},'quoted', \"quoted\", 'a,b,c'}",
+    "http//domain.com:${port}",
+    "${foo_${bar}:arg}",
+    "true${a}",
+    "${foo:\tbar,\tbaz}",
+    "[${foo}, ${bar}]",
+    "${foo:{}}",
+    "{null: [0, 3.14, false]}",
+    "{'a': 0}",
+    "${foo:{'a': 0}}",
+    "{${foo}: ${bar}}",
+    "${foo:{${bar}: 0, ${baz}: 1}, 2}_abc",
+    "{ab_${foo}: c_${baz}}",
```

Comment:
I think this is too much at this point, I am not expecting people to register so many functions that this is needed.

Reply:
Noted

Reply:
I'm doing a pass through all unresolved conversations (I'll let you resolve them when you are good).
The syntax `${ab_${foo}:...}` is not allowed anymore by the grammar, so this should be ok now.